Windows: test_tools.py failures on Python 3.14+

`test_tools.py` starting to fail, on Windows, since Python 3.14 ([example run](https://github.com/BoboTiG/python-mss/actions/runs/19522485113/job/55888581962?pr=426)).

I guess this is due to "[gh-91349](https://github.com/python/cpython/issues/91349): Replaces our copy of zlib with zlib-ng, for performance improvements in [zlib](https://docs.python.org/3.14/library/zlib.html#module-zlib)" on the CPython side. In `tools.to_png()` where `zlib` is used to compress data chunks, the new zlib implementation differs in some way.

To reproduce also on Linux, I installed [python-zlib-ng](https://github.com/pycompression/python-zlib-ng) and use it in `tools.py` like that: `from zlib_ng import zlib_ng as zlib`.

Same result: all tests fail in `test_tools.py`.

Example:

```
=========================== short test summary info ===========================
FAILED src/tests/test_tools.py::test_compression_level[default] - AssertionError: assert 'cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41' == 'ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5'
  
  - ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5
  + cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41
FAILED src/tests/test_tools.py::test_compression_levels[1-841665ec73b641dfcafff5130b497f5c692ca121caeb06b1d002ad3de5c77321] - AssertionError: assert '9b092af99bc4cf97fd998690e6641d5a1647e02124ef366220009aebb72d2cf9' == '841665ec73b641dfcafff5130b497f5c692ca121caeb06b1d002ad3de5c77321'
  
  - 841665ec73b641dfcafff5130b497f5c692ca121caeb06b1d002ad3de5c77321
  + 9b092af99bc4cf97fd998690e6641d5a1647e02124ef366220009aebb72d2cf9
FAILED src/tests/test_tools.py::test_compression_levels[2-b11107163207f68f36294deb3f8e6b6a5a11399a532917bdd59d1d5f1117d4d0] - AssertionError: assert 'b328358258fd80c3669b3648279be3d898b2d20611507ae10d7c569c20838816' == 'b11107163207f68f36294deb3f8e6b6a5a11399a532917bdd59d1d5f1117d4d0'
  
  - b11107163207f68f36294deb3f8e6b6a5a11399a532917bdd59d1d5f1117d4d0
  + b328358258fd80c3669b3648279be3d898b2d20611507ae10d7c569c20838816
FAILED src/tests/test_tools.py::test_compression_levels[3-31278bad8c1c077c715ac4f3b497694a323a71a87c5ff8bdc7600a36bd8d8c96] - AssertionError: assert 'a66f4e438843c72de1530f301e3ebc57bd747a427236249f57ea5f44de5c9db0' == '31278bad8c1c077c715ac4f3b497694a323a71a87c5ff8bdc7600a36bd8d8c96'
  
  - 31278bad8c1c077c715ac4f3b497694a323a71a87c5ff8bdc7600a36bd8d8c96
  + a66f4e438843c72de1530f301e3ebc57bd747a427236249f57ea5f44de5c9db0
FAILED src/tests/test_tools.py::test_compression_levels[4-8f7237e1394d9ddc71fcb1fa4a2c2953087562ef6eac85d32d8154b61b287fb0] - AssertionError: assert '7525470a876abe2c2ae14a36224e19f1e61c999171967743e6c822518661a00d' == '8f7237e1394d9ddc71fcb1fa4a2c2953087562ef6eac85d32d8154b61b287fb0'
  
  - 8f7237e1394d9ddc71fcb1fa4a2c2953087562ef6eac85d32d8154b61b287fb0
  + 7525470a876abe2c2ae14a36224e19f1e61c999171967743e6c822518661a00d
FAILED src/tests/test_tools.py::test_compression_levels[5-83a55f161bad2d511b222dcd32059c9adf32c3238b65f9aa576f19bc0a6c8fec] - AssertionError: assert 'f96785da9f8783eb3d3fd8b4913b9bc9dce34ee3a766db3822f3b2bc21c0ce6d' == '83a55f161bad2d511b222dcd32059c9adf32c3238b65f9aa576f19bc0a6c8fec'
  
  - 83a55f161bad2d511b222dcd32059c9adf32c3238b65f9aa576f19bc0a6c8fec
  + f96785da9f8783eb3d3fd8b4913b9bc9dce34ee3a766db3822f3b2bc21c0ce6d
FAILED src/tests/test_tools.py::test_compression_levels[6-ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5] - AssertionError: assert 'cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41' == 'ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5'
  
  - ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5
  + cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41
FAILED src/tests/test_tools.py::test_compression_levels[7-85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac] - AssertionError: assert '15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2' == '85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac'
  
  - 85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac
  + 15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2
FAILED src/tests/test_tools.py::test_compression_levels[8-85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac] - AssertionError: assert '15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2' == '85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac'
  
  - 85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac
  + 15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2
FAILED src/tests/test_tools.py::test_compression_levels[9-85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac] - AssertionError: assert '15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2' == '85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac'
  
  - 85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac
  + 15ce862cfab5d6fe0dd5787e0253e3e488f2b3e2b5cae23c053858b91fdc6ad2
FAILED src/tests/test_tools.py::test_output_file - AssertionError: assert 'cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41' == 'ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5'
  
  - ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5
  + cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41
FAILED src/tests/test_tools.py::test_output_raw_bytes - AssertionError: assert 'cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41' == 'ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5'
  
  - ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5
  + cada71e84da51247febaa56ca261836af243f84b11c14e18053242b131611b41
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -140,7 +140,7 @@ addopts = """
   --showlocals
   --strict-markers
   -r fE
-  -vvv
+  -v
   --cov=src/mss
   --cov-report=term-missing:skip-covered
 """

--- a/src/tests/test_tools.py
+++ b/src/tests/test_tools.py
@@ -2,70 +2,58 @@
 Source: https://github.com/BoboTiG/python-mss.
 """
 
-import hashlib
-import zlib
-from collections.abc import Callable
+from __future__ import annotations
+
+import io
 from pathlib import Path
+from typing import TYPE_CHECKING
 
 import pytest
 
-from mss.base import MSSBase
 from mss.tools import to_png
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from mss.base import MSSBase
 
 WIDTH = 10
 HEIGHT = 10
-MD5SUM = "ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5"
+
+
+def assert_is_valid_png(*, raw: bytes | None = None, file: Path | None = None) -> None:
+    Image = pytest.importorskip("PIL.Image", reason="PIL module not available.")  # noqa: N806
+
+    assert bool(Image.open(io.BytesIO(raw) if raw is not None else file).tobytes())
+    try:
+        Image.open(io.BytesIO(raw) if raw is not None else file).verify()
+    except Exception:  # noqa: BLE001
+        pytest.fail(reason="invalid PNG data")
 
 
 def test_bad_compression_level(mss_impl: Callable[..., MSSBase]) -> None:
-    with mss_impl(compression_level=42) as sct, pytest.raises(zlib.error):
+    with mss_impl(compression_level=42) as sct, pytest.raises(Exception, match="Bad compression level"):
         sct.shot()
 
 
-def test_compression_level(mss_impl: Callable[..., MSSBase]) -> None:
-    data = b"rgb" * WIDTH * HEIGHT
-    output = Path(f"{WIDTH}x{HEIGHT}.png")
-
-    with mss_impl() as sct:
-        to_png(data, (WIDTH, HEIGHT), level=sct.compression_level, output=output)
-
-    assert hashlib.sha256(output.read_bytes()).hexdigest() == MD5SUM
-
-
-@pytest.mark.parametrize(
-    ("level", "checksum"),
-    [
-        (0, "547191069e78eef1c5899f12c256dd549b1338e67c5cd26a7cbd1fc5a71b83aa"),
-        (1, "841665ec73b641dfcafff5130b497f5c692ca121caeb06b1d002ad3de5c77321"),
-        (2, "b11107163207f68f36294deb3f8e6b6a5a11399a532917bdd59d1d5f1117d4d0"),
-        (3, "31278bad8c1c077c715ac4f3b497694a323a71a87c5ff8bdc7600a36bd8d8c96"),
-        (4, "8f7237e1394d9ddc71fcb1fa4a2c2953087562ef6eac85d32d8154b61b287fb0"),
-        (5, "83a55f161bad2d511b222dcd32059c9adf32c3238b65f9aa576f19bc0a6c8fec"),
-        (6, "ee1b645cc989cbfc48e613b395a929d3d79a922b77b9b38e46647ff6f74acef5"),
-        (7, "85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac"),
-        (8, "85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac"),
-        (9, "85f8d1b01cef926c111b194229bd6c01e2a65b18b4dd902293698e6de8f4e9ac"),
-    ],
-)
-def test_compression_levels(level: int, checksum: str) -> None:
+@pytest.mark.parametrize("level", range(10))
+def test_compression_level(level: int) -> None:
     data = b"rgb" * WIDTH * HEIGHT
     raw = to_png(data, (WIDTH, HEIGHT), level=level)
     assert isinstance(raw, bytes)
-    sha256 = hashlib.sha256(raw).hexdigest()
-    assert sha256 == checksum
+    assert_is_valid_png(raw=raw)
 
 
 def test_output_file() -> None:
     data = b"rgb" * WIDTH * HEIGHT
     output = Path(f"{WIDTH}x{HEIGHT}.png")
     to_png(data, (WIDTH, HEIGHT), output=output)
-
     assert output.is_file()
-    assert hashlib.sha256(output.read_bytes()).hexdigest() == MD5SUM
+    assert_is_valid_png(file=output)
 
 
 def test_output_raw_bytes() -> None:
     data = b"rgb" * WIDTH * HEIGHT
     raw = to_png(data, (WIDTH, HEIGHT))
     assert isinstance(raw, bytes)
-    assert hashlib.sha256(raw).hexdigest() == MD5SUM
+    assert_is_valid_png(raw=raw)

